### PR TITLE
Short types with ignore ref as level

### LINKED ISSUE #1613: Short hint type
Ignore ref as level. 
For example:
```rust
let option/*: &Option<&_> */ = &Some(&String::new());
let error/*: Option<Result<u64, E>> */ = Some(Ok(0u64));
```

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #1613

### CODE REVIEW

**Comment by farodin91** [comment]
@matklad r?

File: src/test/kotlin/org/rust/lang/core/type/RsShortTypeInferenceTest.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            let s: S<(), ()> = unimplemented!();
+            let foo = s
+                .wrap(|x: i32| x + 1)
+                .wrap(|x: i32| x + 1)
+                .wrap(|x: i32| x + 1)
+                .wrap(|x: i32| x + 1);
+            foo;
+            //^ S<fn() -> <unknown>, S<fn() -> <unknown>, S<_, _>>>
+        }
+    """, short = true)
```
Comment (matklad) [inline src/test/kotlin/org/rust/lang/core/type/RsShortTypeInferenceTest.kt]: Let's use hint tests to check this behavior? Typification tests are about what types are actually inferred, and hint tests are more presentation oriented.

Comment (farodin91) [inline src/test/kotlin/org/rust/lang/core/type/RsShortTypeInferenceTest.kt]: I want to split this from the logic of hinting

Comment (matklad) [inline src/test/kotlin/org/rust/lang/core/type/RsShortTypeInferenceTest.kt]: Seems reasonable! Then let's add a test to `org.rust.lang.core.ext.RsPsiStructureTest`, which allows you to assert about the outcome of arbitrary function of PSI element. This way, we won't tie presentation and type inference tests.

Comment (farodin91) [inline src/test/kotlin/org/rust/lang/core/type/RsShortTypeInferenceTest.kt]: Done

**Pull request merged**